osquery/tables/yara/yara.cpp

```diff
@@ -97,6 +104,57 @@ static YARAConfigParser getYaraParser(void) {
   return yaraParser;
 }
 
+bool isRuleUrlAllowed(std::set<std::string> signature_set, std::string url) {
+  Uri test_uri(url);
+  for (const auto& sig : signature_set) {
+    Uri sig_uri(sig);
+
+    // The uri scheme, host and path matches are case sensitive
+    if ((sig_uri.host() == test_uri.host()) &&
+        (sig_uri.scheme() == test_uri.scheme())) {
+      // Check the regex pattern for the allowed URL
+      const std::regex pat(sig_uri.path());
+      if (std::regex_match(test_uri.path(), pat)) {
+        return true;
+      }
+    }
+  }
+  return false;
+}
+
+Status getRuleFromURL(const std::string& url, std::string& rule) {
+  auto yaraParser = getYaraParser();
+  if (isNull(yaraParser)) {
+    return Status::failure("YARA config parser plugin not found");
+  }
+
+  try {
+    auto signature_set = yaraParser->url_allow_set();
+    if (!isRuleUrlAllowed(signature_set, url)) {
+      VLOG(1) << "YARA signature url " << url << " not allowed";
+      return Status::failure("YARA signature url not allowed");
+    }
+
+    http::Client client(TLSTransport().getOptions());
```

Comment:
I recommend `getInternalOptions` as it will apply more restrictive requirements, for example limiting the cipher suites and optionally applying a client cert if it is used. It is our goal to use `getInternalOptions` when communicating with osquery-related infra such as logging and configuration endpoints / carver endpoints / and similarly I think YARA signature endpoints should apply as well.

Reply:
Thanks, @theopolis for suggestion and providing details about `getInternalOptions`. I updated it accordingly.